[UnifiedPDF] PDFPresentationController::updateForCurrentScrollability() should use the scrollability provided by callers

aprotyas opened the pull request:
#### 7f5aad034fe5a856ca4209d2b2dfa26efa9e3ddf
<pre>
[UnifiedPDF] PDFPresentationController::updateForCurrentScrollability() should use the scrollability provided by callers
<a href="https://bugs.webkit.org/show_bug.cgi?id=277533">https://bugs.webkit.org/show_bug.cgi?id=277533</a>
<a href="https://rdar.apple.com/133042660">rdar://133042660</a>

Reviewed by Tim Horton.

Currently, the discrete controller does not observe the updated
scrollability at all, while the scrolling controller asks the plugin to
compute scrollability, which is a duplication of work.

This patch addresses both of these issues.

Finally, we make UnifiedPDFPlugin::computeScrollability() private, since
this should not be called by the presentation controller.

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/PDFDiscretePresentationController.mm:
(WebKit::PDFDiscretePresentationController::updateForCurrentScrollability):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/PDFScrollingPresentationController.mm:
(WebKit::PDFScrollingPresentationController::updateForCurrentScrollability):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::didChangeVisibleRow):

Ensure that the presentation controller accounts for scrollability of
the newly visible row.

Canonical link: <a href="https://commits.webkit.org/281775@main">https://commits.webkit.org/281775@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/53274893431bd05771658dd9a5e0b23b54c00622

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60936 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/40295 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/13512 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64867 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/11483 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47971 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11758 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/49267 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7972 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62970 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/37509 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52795 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/30092 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/34192 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/10396 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/56018 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/10310 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66598 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/4880 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/10139 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/56636 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4902 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52753 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/56826 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/4042 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9175 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/36100 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/37182 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/38275 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/36927 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/PDFDiscretePresentationController.mm:
```diff
@@ -1306,9 +1306,12 @@ static float elasticDeltaForTimeDelta(float initialPosition, float initialVeloci
+    for (auto& row : m_rows) {
+        if (auto* tiledBacking = row.contentsLayer->tiledBacking())
+            tiledBacking->setScrollability(scrollability);
+    }
```
@smfr does it make sense to update this for all rows? Or only for the visible row?

smfr replied:
Only the visible row, but when another row becomes visible you'll need to update its tiledBacking.

aprotyas replied:
In [5327489](https://github.com/WebKit/WebKit/commit/53274893431bd05771658dd9a5e0b23b54c00622), I only update scrollability for the visible row. I also add a call to updateForCurrentScrollability() in UnifiedPDFPlugin::didChangeVisibleRow() to update the tiled backing of the new visible row.

aprotyas committed:
[UnifiedPDF] PDFPresentationController::updateForCurrentScrollability() should use the scrollability provided by callers
https://bugs.webkit.org/show_bug.cgi?id=277533
rdar://133042660

Reviewed by Tim Horton.

Currently, the discrete controller does not observe the updated
scrollability at all, while the scrolling controller asks the plugin to
compute scrollability, which is a duplication of work.

This patch addresses both of these issues.

Finally, we make UnifiedPDFPlugin::computeScrollability() private, since
this should not be called by the presentation controller.

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/PDFDiscretePresentationController.mm:
(WebKit::PDFDiscretePresentationController::updateForCurrentScrollability):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/PDFScrollingPresentationController.mm:
(WebKit::PDFScrollingPresentationController::updateForCurrentScrollability):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::didChangeVisibleRow):

Ensure that the presentation controller accounts for scrollability of
the newly visible row.

Canonical link: https://commits.webkit.org/281775@main
